de.dlr.sc.virsat.model.extension.mechanical/src/de/dlr/sc/virsat/model/extension/mechanical/catia/CatiaExporter.java

```diff
@@ -0,0 +1,77 @@
+/*******************************************************************************
+ * Copyright (c) 2008-2019 German Aerospace Center (DLR), Simulation and Software Technology, Germany.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package de.dlr.sc.virsat.model.extension.mechanical.catia;
+
+import java.util.Collection;
+import java.util.Collections;
+
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+
+import de.dlr.sc.virsat.model.concept.types.structural.IBeanStructuralElementInstance;
+import de.dlr.sc.virsat.model.extension.ps.model.ConfigurationTree;
+import de.dlr.sc.virsat.model.extension.visualisation.model.Visualisation;
+
+/**
+ * This class creates the JSON representation of a product structure tree.
+ *
+ */
+
+public class CatiaExporter {
+	
+	/**
+	 * Main method that creates the JSON representation of a configuration tree
+	 * @param configurationTree the configuration tree
+	 * @return the json root object
+	 */
+	@SuppressWarnings("unchecked")
+	public JSONObject transform(ConfigurationTree configurationTree) {
+		JSONObject json = new JSONObject();
+		
+		// TODO Get the Element definitions
+		json.put(CatiaProperties.PARTS, transformParts(Collections.emptySet())); 
+		
+		return json;
+	}
+	
+	/**
+	 * Creates the JSON representation for a collection of parts
+	 * @param parts the parts to transform
+	 * @return the json representation
+	 */
+	@SuppressWarnings("unchecked")
+	public JSONArray transformParts(Collection<? extends IBeanStructuralElementInstance> parts) {
+		JSONArray jsonParts = new JSONArray();
+		
+		for (IBeanStructuralElementInstance part : parts) {
+			Visualisation vis = part.getFirst(Visualisation.class);
+			
+			if (vis != null) {
+				JSONObject jsonPart = transformElement(part);
+				jsonParts.add(jsonPart);
+			}
+		}
+		
+		return jsonParts;
+	}
+	
+	/**
+	 * Creates a JSON object for a single virtual satellite element
+	 * @param element the element definition
+	 * @return the JSON object
+	 */
+	@SuppressWarnings("unchecked")
+	public JSONObject transformElement(IBeanStructuralElementInstance element) {

```

Comment:
The Visualization eban could be part of the methods signature? I think it will be needed later anyway<

Reply:
The idea was later to have methods like "transformCircle", "transformShape" etc.
This method should only take care of creating json object and putting in the identifiers name and uuid. All other methods that add more stuff should call this base method. So we dont repeat putting name and uuid into the json objects we create.